contributors/devel/pull-requests.md

```diff
@@ -77,35 +96,255 @@ $ $RELNOTES -man # for details on how to use the tool
 $ $RELNOTES --branch=master
 ```
 
-## Visual overview
+Release notes apply to PRs on the master branch. For cherry-pick PRs, see the [cherry-pick instructions](cherry-picks.md). The only exception to these rules is when a PR is not a cherry-pick and is targeted directly to the non-master branch.  In this case, a `release-note-*` label is required for that non-master PR.
+
+**Labels**
+
+1. All pull requests are initiated with a `release-note-label-needed` label if you don't specify them in your original PR. If you are a new contributor you won't have access to modify labels, instead leave a comment as instructed below or ask in your original PR. 
+
+1. Remove the `release-note-label-needed` label and replace it with one of the other `release-note-*` labels:
+    1. `release-note-none` is a valid option if the PR does not need to be mentioned at release time
+    1. `release-note` labeled PRs generate a release note using the PR title by default OR the release-note block in the PR template, if it's filled in
+    
+**Comments**
+
+Or, commenting either `/release-note` or `/release-note-none` will also set the `release-note` or `release-note-none` labels respectively.
+
+Now that your release notes are in shape, let's look at how the PR gets tested and merged.
+
+## The Testing and Merge Workflow
+
+The Kubernetes merge workflow uses comments to run tests and labels for merging PRs. NOTE: For pull requests that are in progress but not ready for review, prefix the PR title with "WIP" and track any remaining TODOs in a checklist in the pull request description.
+
+Here's the process the PR goes through on its way from submission to merging.
+
+1. Make the pull request
+1. mergebot assigns a reviewer
+
+If you're **not** a member:
+
+1. Reviewer suggests edits
+1. Push edits to your PR branch
+1. Repeat the prior two steps as needed
+1. (Optional) Some reviewers prefer that you squash commits at this step. 
+1. Reviewer adds the labels `lgtm` and `ok-to-merge` which triggers the final tests and merge process (this can be done with the command `/lgtm`), or, if they are a member, can add the comment `@k8s-bot ok to test` which triggers the intermediate tests
+
+If you are in the not a member, or a member comments `@k8s-bot ok to test`, the intermediate tests will run:
+
+1. Automatic tests run
+    1. travis (go build/fmt and generated docs)
+    1. Jenkins GCE e2e (minimal e2e suite on GCE)
+    1. Jenkins unit/integration (unit and local cluster tests)
+1. If tests fail, resolve issues by pushing edits to your PR branch
+1. If the failure is a flake, a member can comment `@k8s-bot [e2e|unit] test this issue: #<flake issue>`
+
+Once the tests pass, all failures are commented as flakes, or the reviewer adds the labels `lgtm` and `ok-to-merge`, the PR enters the final merge queue. The merge queue is needed to make sure no incompatible changes have been introduced by other PRs since the tests were last run on your PR.
```

Comment:
`ok-to-merge` doesn't exist, same as above. `approved` is also needed before PR enters submit queue